src/Illuminate/Http/Request.php

```diff
@@ -744,7 +744,8 @@ public function offsetExists($offset)
 	 */
 	public function offsetGet($offset)
 	{
-		return $this->input($offset);
+		$all = $this->all();
+		return $all[$offset];
```

Comment:
Just do `return $this->all()[$offset];`.

It works on php 5.4+, so that's fine. http://3v4l.org/sve0Z